.travis.yml

```diff
@@ -1,18 +1,15 @@
 rvm: 2.4.1
 cache: bundler
-sudo: false
+sudo: required
 dist: trusty
-addons:
-  apt:
-    sources:
-    - elasticsearch-5.x
-    packages:
-    - elasticsearch
+before_install:
+- echo "deb https://artifacts.elastic.co/packages/6.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-6.x.list
+- sudo apt-get install elasticsearch
+- sudo service elasticsearch start; sleep 10
 before_script:
 - cp test/dummy/.env.example test/dummy/.env
 - curl localhost:9200
 - bundle exec rake app:db:setup
 - bundle exec rake app:index:reset
 services:
-- elasticsearch
 - postgresql
```

Comment:
You could add something like this in the `before_script`, which replaces uses of both `sleep` and `curl`:

https://serverfault.com/questions/42021/how-to-ping-in-linux-until-host-is-known

ping defaults to 10s but that can be changed to 1s.